components/MetadataCreation/LinkPreview.tsx

```diff
@@ -1,11 +1,23 @@
+import { useState, useEffect } from "react";
 import { useMomentCreateProvider } from "@/providers/MomentCreateProvider/MomentCreateProvider";
 import LinkInput from "./LinkInput";
 import Image from "next/image";
 import { useMomentFormProvider } from "@/providers/MomentFormProvider";
 
 const LinkPreview = () => {
   const { createdContract } = useMomentCreateProvider();
-  const { previewSrc, link } = useMomentFormProvider();
+  const { previewFile, link } = useMomentFormProvider();
+  const [previewFileUrl, setPreviewFileUrl] = useState<string>("");
+
+  useEffect(() => {
+    if (previewFile) {
+      const blobUrl = URL.createObjectURL(previewFile);
+      setPreviewFileUrl(blobUrl);
+      return () => URL.revokeObjectURL(blobUrl);
+    } else {
+      setPreviewFileUrl("");
+    }
+  }, [previewFile]);
```

Comment:
Open Closed Principle
- actual: new useEffects are added to existing react component.
- required: new hook is created and imported into the LinkPreview component.